apis/cache/v1alpha2/replication_group_types.go

```diff
@@ -63,18 +63,201 @@ var LatestSupportedPatchVersion = map[MinorVersion]PatchVersion{
+	// node group (shard).
+	PrimaryEndpoint Endpoint `json:"primaryEndpoint,omitempty"`
+
+	// Slots is the keyspace for this node group (shard).
+	Slots string `json:"slots,omitempty"`
+
+	// Status of this replication group - creating, available, etc.
+	Status string `json:"status,omitempty"`
+}
+
+// NodeGroupMember represents a single node within a node group (shard).
+// Please also see https://docs.aws.amazon.com/goto/WebAPI/elasticache-2015-02-02/NodeGroupMember
+type NodeGroupMember struct {
+	// CacheClusterId is the ID of the cluster to which the node belongs.
+	CacheClusterId string `json:"cacheClusterId,omitempty"`
+
+	// CacheNodeId is the ID of the node within its cluster. A node ID is a
+	// numeric identifier (0001, 0002, etc.).
+	CacheNodeId string `json:"cacheNodeId,omitempty"`
+
+	// CurrentRole is the role that is currently assigned to the node - primary
+	// or replica. This member is only applicable for Redis (cluster mode
+	// disabled) replication groups.
+	CurrentRole string `json:"currentRole,omitempty"`
+
+	// PreferredAvailabilityZone is the name of the Availability Zone in
+	// which the node is located.
+	PreferredAvailabilityZone string `json:"preferredAvailabilityZone,omitempty"`
+
+	// ReadEndpoint is the information required for client programs to connect to a
+	// node for read operations. The read endpoint is only applicable on Redis
+	// (cluster mode disabled) clusters.
+	ReadEndpoint Endpoint `json:"readEndpoint,omitempty"`
+}
+
+// ReplicationGroupPendingModifiedValues are the settings to be applied to the
+// Redis replication group, either immediately or during the next maintenance
+// window. Please also see
+// https://docs.aws.amazon.com/goto/WebAPI/elasticache-2015-02-02/ReplicationGroupPendingModifiedValues
+type ReplicationGroupPendingModifiedValues struct {
+	// AutomaticFailoverStatus indicates the status of Multi-AZ with automatic
+	// failover for this Redis replication group.
+	AutomaticFailoverStatus string `json:"automaticFailoverStatus,omitempty"`
+
+	// PrimaryClusterID that is applied immediately or during the next
+	// maintenance window.
+	PrimaryClusterID string `json:"primaryClusterId,omitempty"`
+
+	// Resharding is the status of an online resharding operation.
+	Resharding ReshardingStatus `json:"resharding,omitempty"`
+}
+
+// The status of an online resharding operation.
+// Please also see https://docs.aws.amazon.com/goto/WebAPI/elasticache-2015-02-02/ReshardingStatus
+type ReshardingStatus struct {
+	// Represents the progress of an online resharding operation.
+	SlotMigration SlotMigration `json:"slotMigration"`
+}
+
+// Represents the progress of an online resharding operation.
+// Please also see https://docs.aws.amazon.com/goto/WebAPI/elasticache-2015-02-02/SlotMigration
+type SlotMigration struct {
+	// NOTE(muvaf): Type of ProgressPercentage is float64 in AWS SDK but
+	// float is not supported by controller-runtime.
```

Comment:
http://giphygifs.s3.amazonaws.com/media/QUaqJRizED5NC/giphy.gif

Reply:
Be careful my friend, knowing too much might get you in trouble..